ziplineApiDump doesn’t work with the configuration cache

```
> Task ziplineApiDump

FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

4 problems were found storing the configuration cache, 2 of which seem unique.
- Task `:compileKotlinJvmZiplineApiDump` of type `app.cash.zipline.gradle.ValidateZiplineApiTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/8.4/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
```

--- a/zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ValidateZiplineApiTask.kt
+++ b/zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ValidateZiplineApiTask.kt
@@ -22,7 +22,6 @@ import javax.inject.Inject
 import okio.Buffer
 import org.gradle.api.DefaultTask
 import org.gradle.api.file.ConfigurableFileCollection
-import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.internal.file.FileCollectionFactory
 import org.gradle.api.provider.Property
@@ -47,6 +46,9 @@ abstract class ValidateZiplineApiTask @Inject constructor(
 
   @get:OutputFile
   abstract val ziplineApiFile: RegularFileProperty
+
+  @get:Input
+  abstract val projectDirectory: Property<String>
 
   @get:Input
   abstract val javaHome: Property<String>
@@ -75,12 +77,13 @@ abstract class ValidateZiplineApiTask @Inject constructor(
   @TaskAction
   fun task() {
     val tomlFile = ziplineApiFile.get().asFile
-    val tomlFileRelative = tomlFile.relativeTo(project.projectDir)
+    val projectDirectory = projectDirectory.get()
+    val tomlFileRelative = tomlFile.relativeTo(File(projectDirectory))
     val queue = workerExecutor.noIsolation()
     queue.submit(ZiplineApiValidatorWorker::class.java) {
       it.cliClasspath.from(cliClasspath)
       it.mode.set(mode)
-      it.projectDirectory.set(project.projectDir)
+      it.projectDirectory.set(projectDirectory)
       it.tomlFile.set(tomlFileRelative)
       it.javaHome.set(File(javaHome.get()))
       it.jdkRelease.set(jdkRelease.get())
@@ -98,7 +101,7 @@ abstract class ValidateZiplineApiTask @Inject constructor(
 private interface ZiplineApiValidatorParameters : WorkParameters {
   val cliClasspath: ConfigurableFileCollection
   val mode: Property<Mode>
-  val projectDirectory: DirectoryProperty
+  val projectDirectory: Property<String>
   val tomlFile: RegularFileProperty
   val javaHome: RegularFileProperty
   val jdkRelease: Property<Int>
@@ -112,7 +115,7 @@ private abstract class ZiplineApiValidatorWorker @Inject constructor(
   override fun execute() {
     val errorOutput = Buffer()
     val execResult = execOperations.javaexec { exec ->
-      val workingDirectory = parameters.projectDirectory.asFile.get()
+      val workingDirectory = File(parameters.projectDirectory.get())
 
       exec.isIgnoreExitValue = true
       exec.classpath = parameters.cliClasspath

--- a/zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt
+++ b/zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt
@@ -180,6 +180,7 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
     task.configure { task ->
       task.cliClasspath.from(cliConfiguration)
       task.ziplineApiFile.set(project.file("api/zipline-api.toml"))
+      task.projectDirectory.set(project.projectDir.path)
 
       // TODO: the validation uses the wrong JDK. We should be getting the JDK from the
       //     KotlinCompile task (as defaultKotlinJavaToolchain.get().buildJvm), but it doesn't

